src/main/java/org/openrewrite/staticanalysis/RemoveUnusedParams.java

```diff
@@ -0,0 +1,158 @@
+                                .collect(Collectors.toSet());
+
+                        // Find which are actually used
+                        Set<String> used = new HashSet<>();
+                        new JavaIsoVisitor<Set<String>>() {
+                            Deque<Set<String>> shadowed = new ArrayDeque<>();
+
+                            @Override
+                            public J.Block visitBlock(J.Block block, Set<String> u) {
+                                shadowed.push(new HashSet<>());
+                                J.Block b = super.visitBlock(block, u);
+                                shadowed.pop();
+                                return b;
+                            }
+
+                            @Override
+                            public J.VariableDeclarations visitVariableDeclarations(J.VariableDeclarations vars, Set<String> u) {
+                                vars.getVariables().forEach(v -> shadowed.peek().add(v.getSimpleName()));
+                                return super.visitVariableDeclarations(vars, u);
+                            }
+
+                            @Override
+                            public J.Identifier visitIdentifier(J.Identifier ident, Set<String> u) {
+                                for (Set<String> scope : shadowed) {
+                                    if (scope.contains(ident.getSimpleName())) {
+                                        return ident;
+                                    }
+                                }
+                                if (params.contains(ident.getSimpleName())) {
+                                    u.add(ident.getSimpleName());
+                                }
+                                return ident;
+                            }
+                        }.visit(m.getBody(), used);
+
+                        // Rebuild parameter list
+                        List<Statement> newParams = new ArrayList<>();
+                        for (Statement p : m.getParameters()) {
+                            if (!(p instanceof J.VariableDeclarations)) {
+                                newParams.add(p);
+                                continue;
+                            }
+                            J.VariableDeclarations vd = (J.VariableDeclarations) p;
+                            if (!vd.getLeadingAnnotations().isEmpty()) {
+                                newParams.add(vd);
+                                continue;
+                            }
+                            List<J.VariableDeclarations.NamedVariable> keep = vd.getVariables().stream()
+                                    .filter(v -> used.contains(v.getSimpleName()))
+                                    .collect(Collectors.toList());
+                            if (!keep.isEmpty()) {
+                                newParams.add(vd.withVariables(keep));
+                            }
```

Comment:
```suggestion
                            newParams.add(vd.withVariables(vd.getVariables().stream()
                                    .filter(v -> used.contains(v.getSimpleName()))
                                    .collect(Collectors.toList())));
```
empty * empty = empty 
empty + empty = empty


- https://github.com/openrewrite/rewrite-static-analysis/pull/558